validphys2/src/validphys/closuretest/multiclosure_nsigma_helpers.py

```diff
@@ -0,0 +1,170 @@
+"""
+This module contains some helper functions that are used for the computation of nsigma in the
+context of a multi-closure test.
+"""
+
+import dataclasses
+import pandas as pd
+import numpy as np
+import logging
+
+from reportengine import collect
+
+from validphys.core import DataSetSpec, PDF
+from nnpdf_data.coredata import CommonData
+from validphys.calcutils import calc_chi2
+from validphys import convolution
+
+log = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class CentralChi2Data:
+    value: float
+    ndata: int
+    dataset: DataSetSpec
+
+    @property
+    def reduced(self):
+        return self.value / self.ndata
+
+
+def central_predictions(dataset: DataSetSpec, pdf: PDF) -> pd.DataFrame:
+    """
+    Computes the central prediction (central PDF member) for a dataset.
+
+    Parameters
+    ----------
+    dataset: validphys.core.DataSetSpec
+    pdf: validphys.core.PDF
+
+    Returns
+    -------
+    pd.DataFrame
+        index is datapoints, column is the central prediction.
+    """
+    return convolution.central_predictions(dataset, pdf)
+
+
+def central_member_chi2(
+    central_predictions: pd.DataFrame,
+    sqrt_covmat: np.ndarray,
+    dataset: DataSetSpec,
+    loaded_commondata_with_cuts: CommonData,
+) -> CentralChi2Data:
+    """
+    Computes the chi2 value for a dataset.
+
+    Parameters
+    ----------
+    central_predictions:
+        The central predictions for the dataset.
+    sqrt_covmat: np.ndarray
+        The square root of the covariance matrix.
+    dataset: DataSetSpec
+        The dataset.
+    loaded_commondata_with_cuts: nnpdf_data.coredata.CommonData
+
+    Returns
+    -------
+    CentralChi2Data
+    """
+    diff = (
+        central_predictions.values.ravel()
+        - loaded_commondata_with_cuts.central_values.values.ravel()
+    )
+
+    value = calc_chi2(sqrt_covmat, diff)
+    ndata = len(central_predictions)
+    return CentralChi2Data(value=value, ndata=ndata, dataset=dataset)
```

Comment:
Test it because I'm not 100% sure (maybe you are doing something in the multiclosure that will break this) but _in principle_ this function could just depend on `abs_chi2_data` which would automatically get the predictions, the data and the compute the chi2.